table/table.go

```diff
@@ -101,108 +92,370 @@ func DefaultKeyMap() KeyMap {
+	m.table.
+		BorderTop(top).
+		BorderRight(right).
+		BorderBottom(bottom).
+		BorderLeft(left).
+		BorderRow(rowSeparator).
+		BorderColumn(columnSeparator)
+	return m
+}
 
-// New creates a new model for the table widget.
-func New(opts ...Option) Model {
-	m := Model{
-		cursor:   0,
-		viewport: viewport.New(viewport.WithHeight(20)), //nolint:mnd
+// Border sets the kind of border to use for the table. See [lipgloss.Border].
+func (m *Model) Border(border lipgloss.Border) *Model {
+	m.styles.border = border
+	m.table.Border(border)
+	return m
+}
 
-		KeyMap: DefaultKeyMap(),
-		Help:   help.New(),
-		styles: DefaultStyles(),
-	}
+// BorderStyle sets the style for the table border.
+func (m *Model) BorderStyle(style lipgloss.Style) *Model {
+	m.styles.borderStyle = style
+	m.table.BorderStyle(style)
+	return m
+}
 
-	for _, opt := range opts {
-		opt(&m)
-	}
+// BorderBottom sets the bottom border.
+func (m *Model) BorderBottom(v bool) *Model {
+	m.styles.borderBottom = v
+	m.table.BorderBottom(v)
+	return m
+}
 
-	m.UpdateViewport()
+// BorderTop sets the top border.
+func (m *Model) BorderTop(v bool) *Model {
+	m.styles.borderTop = v
+	m.table.BorderTop(v)
+	return m
+}
 
+// BorderLeft sets the left border.
+func (m *Model) BorderLeft(v bool) *Model {
+	m.styles.borderLeft = v
+	m.table.BorderLeft(v)
 	return m
 }
 
-// WithColumns sets the table columns (headers).
-func WithColumns(cols []Column) Option {
-	return func(m *Model) {
-		m.cols = cols
-	}
+// BorderRight sets the right border.
+func (m *Model) BorderRight(v bool) *Model {
+	m.styles.borderRight = v
+	m.table.BorderRight(v)
+	return m
 }
 
-// WithRows sets the table rows (data).
-func WithRows(rows []Row) Option {
+// BorderColumn sets the column border.
+func (m *Model) BorderColumn(v bool) *Model {
+	m.styles.borderColumn = v
+	m.table.BorderColumn(v)
+	return m
+}
+
+// BorderHeader sets the header border.
+func (m *Model) BorderHeader(v bool) *Model {
+	m.styles.borderHeader = v
+	m.table.BorderHeader(v)
+	return m
+}
+
+// BorderRow sets the row borders.
+func (m *Model) BorderRow(v bool) *Model {
+	m.styles.borderRow = v
+	m.table.BorderRow(v)
+	return m
+}
+
+// Options
+
+// Option is used to set options in [New]. For example:
+//
+//	table := New(WithHeaders([]string{"Rank", "City", "Country", "Population"}))
+type Option func(*Model)
+
+// WithHeaders sets the table headers. This function is used as an [Option] in
+// when creating a table with [New].
+func WithHeaders(headers ...string) Option {
 	return func(m *Model) {
-		m.rows = rows
+		m.SetHeaders(headers...)
 	}
 }
 
-// WithHeight sets the height of the table.
+// TODO andrey confirm this... I'm pretty sure that's how it's working now
+//
+// WithHeight sets the height of the table. The given height will be the total
+// table height including borders, margins, and padding. This function is used
+// as an [Option] in when creating a table with [New].
```

Comment:
I just made some quick tests, and it is still the full table size, and not the number of rows. It includes borders and white spaces.

I found a bug in the process, though. If less then 5, lipgloss is still allow the table to take 5 lines. I added a new topic on https://github.com/charmbracelet/glamour/issues/416 about this to fix soon.

Reply:
@andreynering I believe 5 is the minimum height for the table as that would be top border + header + header border + "..." + bottom border. That's by design, but we can change that if you have a preferred behaviour in mind

Reply:
Yeah, I understand the reason. 5 is the minimum to have a minimally presentable table.

That said, if the user _really_ asked for a smaller table, I feel that we should cut the table. That's we do horizontally, btw. We have a test for width 1 on lipgloss and it looks like this:

```
┌
│
├
│
│
│
│
│
└
```

Perhaps on the TUI app the space is shared with other components, and if the window is too small, it should do what's asked.